src/Obj.h

```diff
@@ -22,9 +25,48 @@ class Location final {
     bool operator==(const Location& l) const;
     bool operator!=(const Location& l) const { return ! (*this == l); }
 
+    auto FileName() const { return filename; }
+    auto FirstLine() const { return first_line; }
+    auto LastLine() const { return last_line; }
+
+    std::string FileAndLine() const { return std::string(filename) + ":" + std::to_string(first_line); }
+    std::string FileAndLastLine() const { return std::string(filename) + ":" + std::to_string(last_line); }
+    std::string FileAndLines() const {
```

Comment:
Prefer to have these not in the Location class's API. They are helpers for formatting, maybe just inline helpers further down?

```
std::string loc_file_and_line(Location& l) { ... }
```

Though `FileAndLastline()` is only used in the non-functional-never-used debug logger, so not excited having a helper just for that code even in `zeek::detail` (maybe a helper in zeek/Debug.h though?). `FileAndLines()` is used once in Trigger.cc - would just open-code it there. Last, `FileAndLine()` is mostly used in format-strings and seems that could be replaced with `"%s:%d, loc->FileName().c_str(), `loc->FirstLine()`. Seems fine to me even if a bit longer. So yeah, maybe none of those ;-)

Reply:
I just open-coded and removed the helpers. Seemed sensible.